### DISCUSSION TITLE
Airflow Tasks Hanging While Extracting Huge Data files

### DISCUSSION THREAD

**Discussion by kgowrisangar** [apache/airflow#24614] [General]
We recently migrated from Airflow 1.x to 2.3.0 for better performance and to use some of the latest features. But after doing so, we're facing few issues like some Airflow tasks are getting stuck at an extraction step while pulling in large files from either an sftp location or extracting records from db2.

Our extractors are mainly written in Python and its works perfectly fine while running them via cli or in a local environment via Pycharm.

Is there any known compatibility issues with any of the Python Libraries? 

We tried to increase the parallelism and the concurrency parameter from the airflow.cfg but still the issue persists. My guess is since a lot of scheduled jobs are running at the same time in our Airflow, the worker pool is getting over used. Is there any mechanism to overcome this or even to optimize ?

Our next try would be to go with Kubernetes to make it a bit more robust. Any help could be appreciated

**Comment by alexkruc** [comment]
I don't know regarding the db2 thing, but there is an open bug for downloading large files from sftp..
https://github.com/apache/airflow/issues/16286
You might follow up on this to see the progress or submit a PR yourself to fix this ;) 
I haven't seen an open bug on DB2, so I can't really comment on this :)

**Reply by potiuk** [reply]
Seems that the paramiko change in #24512 might solve the problem. We are going to have a new provider's release soon - RCs - so anyone commenting here - please pay attention to devlist announcements and issue for testing the new provider and you can verify if the problem gets fixed by it.
